agent/grpc-external/stats_test.go

```diff
@@ -0,0 +1,130 @@
+package external
+
+import (
+	"context"
+	"net"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/agent/grpc-middleware/testutil"
+	"github.com/hashicorp/consul/agent/grpc-middleware/testutil/testservice"
+	"github.com/hashicorp/consul/proto/prototest"
+)
+
+func TestServer_EmitsStats(t *testing.T) {
```

Comment:
This copies the [`TestHandler_EmitStats`](https://github.com/hashicorp/consul/blob/4abad02abdd892a05001280748fa496624d7758f/agent/grpc-internal/stats_test.go#L25) test. I did some deduplication of the helpers by moving things over to `grpc-middleware/testutil`. (I could do more but kind of wanted a sanity check before I go too far.)